lib/files.js

```diff
@@ -10,14 +10,24 @@ var path = require('path');
 var options = require('./args.js');
 var log = require('davlog');
 var verify = require('./verify.js');
+var hooks = require('./hooks');
 
 var putBall = function(info, callback) {
     var localBall = path.join(options.dir, info.path);
-    process.nextTick(function() {
-        mkdirp(path.dirname(localBall), function() {
-            process.nextTick(function() {
-                info.tarball = localBall;
-                verify.verify(info, callback);
+    info.tarball = localBall;
+
+    hooks.tarball(info, callback, function () {
+        process.nextTick(function() {
+            mkdirp(path.dirname(localBall), function() {
```

Comment:
Should this now be `info.tarball` instead of `localBall` so that the hook owner could change that meta in their method and then the system would work off the new data?